### PR TITLE
Make compatible with Collaboration Kit

### PR DESCRIPTION
**Pull request by MatthewVaadin**
## Description

As part of the work to allow Collaboration Kit to work with the Kubernetes Kit session serialization, it was found that K8s Kit has an issue when trying to determine if a bean has been defined for transient fields. If the class is just `Object`, Spring returns a prototype bean.

## Type of change

- [X] Bugfix
- [ ] Feature

### CODE REVIEW

**Review by heruan — CHANGES_REQUESTED** [review]
Better to extract all the formatting changes to a separate PR, so we keep the relevant change on a clean commit.

File: kubernetes-kit-demo/src/main/java/com/vaadin/kubernetes/demo/generator/DataGenerator.java [hunk 1]
```diff
@@ -25,7 +25,7 @@
 public class DataGenerator {
 
     @Bean
-    CommandLineRunner loadData(ContactRepository contactRepository,
+    public CommandLineRunner loadData(ContactRepository contactRepository,
```
Comment (heruan) [inline kubernetes-kit-demo/src/main/java/com/vaadin/kubernetes/demo/generator/DataGenerator.java:28]: Does this need to be public? Bean suppliers are usually kept private.

File: kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/SerializationProperties.java [hunk 2]
```diff
@@ -37,7 +37,7 @@ public class SerializationProperties {
      *
      * @return configuration for transient fields handling.
      */
-    public TransientsProperties getTransients() {
+    TransientsProperties getTransients() {
```
Comment (heruan) [inline kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/SerializationProperties.java:40]: Are we verifying that this is still discovered by Spring and reflected on the metadata for `.properties` files?

**Comment by mcollovati** [comment]
I propose another little change related to `SpringTransientHandler`.
To prevent runtime errors, like the one with the `Object.class` bean detection resolved by this PR, The `matchesPrototype` method could check if the bean name effectively exists before calling `isPrototype` because the latter throws an exception if the bean is not defined.

```java
    private boolean matchesPrototype(String beanName, Object beanDefinition,
            Class<?> fieldValueType) {
        return appCtx.containsBean(beanName) && appCtx.isPrototype(beanName);
    }
```

In addition, as discussed with @MatthewVaadin, prototype beans may cause issues in deserialization:
for example, if there's a prototype `Consumer` bean registered and a class has a transient field of type `Consumer` that is however not injected (e.g. assigned a lambda in the constructor), the handler will potentially detect it as injectable and during deserialization inject an undesired object.
I wonder if we should remove the prototype detection or find a more precise way to detect if the field instance is really a prototype bean.

**Pull request merged**